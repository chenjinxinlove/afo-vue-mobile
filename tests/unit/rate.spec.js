import { shallowMount, mount } from '@vue/test-utils'
import Demo from '../../src/rate/demo'
import Vue from 'vue'
import Rate from '../../src/rate'

describe('Rate.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Rate)
    expect(Vue.component(Rate.name)).toHaveProperty('name', 'VueComponent')
  })
  // 测试能正常显示内容
  it('props', () => {
    const wrapper = shallowMount(Rate, {
      propsData: {
        type: 'primary',
        plain: true
      }
    })
    expect(wrapper.classes()).toEqual(['afo-tag--primaryplain', 'afo-tag'])
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo)).toMatchSnapshot()
  })
})
