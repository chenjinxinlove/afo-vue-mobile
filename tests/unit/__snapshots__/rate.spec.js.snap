// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate.vue demo snapshot 1`] = `
<div>
  <p>默认显示</p>
  <afo-rate></afo-rate>
  <div style="height: 50px;"></div>
  <p>自定义显示</p>
  <afo-rate size="25" count="6" color="#2ba" void-color="#ceefe8"></afo-rate>
  <div style="height: 50px;"></div>
  <p>禁操作</p>
  <afo-rate disabled=""></afo-rate>
</div>
`;
